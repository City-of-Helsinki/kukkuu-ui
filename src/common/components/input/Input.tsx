import React, { FunctionComponent, ReactElement } from 'react';
import classnames from 'classnames';

import styles from './input.module.scss';

type InputElementAttributes = React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
>;

interface InputProps extends InputElementAttributes {
  type: string;
  label: string | ReactElement | undefined;
  id: string;
}
const Input: FunctionComponent<InputProps> = ({
  type = 'text',
  className,
  label,
  required,
  id,
  ...rest
}) => {
  return (
    <div className={classnames(styles.inputWrapper, styles[`${type}Input`])}>
      <div className={styles.inputTypeWrapper}>
        {label && <label htmlFor={id}>{required ? `${label}*` : label}</label>}
        <input type={type} id={id} {...rest} />
      </div>
    </div>
  );
};

export default Input;
