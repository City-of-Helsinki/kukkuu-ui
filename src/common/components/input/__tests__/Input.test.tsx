import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input from '../Input';

test('renders snapshot correctly', () => {
  const input = shallow(<Input id="foo" label="bar" />);
  expect(toJson(input)).toMatchSnapshot();
});

test('render label with asterisk if required props is true', () => {
  const wrapper = shallow(<Input id="foo" label="bar" required={true} />);
  expect(wrapper.find('label').text()).toContain('*');
});
