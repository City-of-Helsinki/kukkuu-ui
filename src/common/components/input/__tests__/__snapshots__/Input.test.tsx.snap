// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<div class=\\"className inputWrapper textInput\\"><label for=\\"foo\\">bar</label><input type=\\"text\\" id=\\"foo\\"/><small class=\\"inputError\\"></small></div>"`;
